src/Database/LSMTree/Internal/Index.hs

```diff
@@ -95,7 +70,7 @@ class Index i where
         (Currently, construction of compact indexes needs the former and
         construction of ordinary indexes needs the latter bound.)
 -}
-class Index (ResultingIndex j) => IndexAcc j where
+class (Index (ResultingIndex j), NFData (ResultingIndex j)) => IndexAcc j where
```

Comment:
What do you think about replacing this constraint on the class with a constraint in the `SomeIndexAcc` existential? For example:

```haskell
data SomeIndexAcc s = forall j . (IndexAcc j, NFData (ResultingIndex j)) => SomeIndexAcc (j s)
```